### PR TITLE
Ticket/860 public use block

### LINKED ISSUE #546: Public Use configuration error
I believe the fix for #503 is incorrect.

In `cgov_core`, `field.storage.node.field_public_use.yml` defines `public_use` as type `list_integer`.

In `cgov_article` and `cgov_press_release`, the `field.field.node.MODULE.field_public_use.yml` files set field_type to `list_string`. It should be set to `list_integer` instead.

Making a similar change in `pdq_cancer_information_summary` works in the user interface, but Functional tests fail with a schema error.
```
1) Drupal\Tests\cgov_site\Functional\PDQSummaryTest::testSummaryFieldsExist
Drupal\Core\Config\Schema\SchemaIncompleteException: Schema errors for field.field.node.pdq_cancer_information_summary.field_public_use with the following errors: field.field.node.pdq_cancer_information_summary.field_public_use:default_value.0.value variable type is integer but applied schema class is Drupal\Core\TypedData\Plugin\DataType\StringData
```

### LINKED ISSUE #860: Create content block for Public Use Text
In order for the public use text to display in a correct region on content types when the checkbox is selected, we need to also create the reusable content block so that when we install core, this content is automatically created.  

##English Content:
If you would like to reproduce some or all of this content, see Reuse of NCI Information for guidance about copyright and permissions. In the case of permitted digital reproduction, please credit the National Cancer Institute as the source and link to the original NCI product using the original product's title; e.g., “<entity/node page title> was originally published by the National Cancer Institute”

##Spanish Content:
Si desea copiar algo de este texto, vea Derechos de autor y uso de imágenes y contenido sobre instrucciones de derechos de autor y permisos. En caso de reproducción digital permitida, por favor, dé crédito al Instituto Nacional del Cáncer como su creador, y enlace al producto original del NCI usando el título original del producto; por ejemplo, “<entity/node page title> publicada originalmente por el Instituto Nacional del Cáncer”.

On Install, the process should be:

- [ ] Create the content block for the Public Reuse text
- [ ] Add the block to region on Articles, blog posts, etc.
- [ ] Have the template display the public reuse text when public reuse text is checked

### PR DESCRIPTION
**Pull request by bgnih**
Fixes #860 Fixes #546 Add Public Use Block.  Adds Token_Filter module (with my patch from Drupal.org) to display current page title in block.  Adds the token filter option to steamlined, basic html and full html text filters.    Creates the Public Use block in cgov_core.install and adds the translated version(s) in cgov_core.module:cgov_core_themes_installed() .  Adds public_use region for displaying the public_use block within.  Also fixes the changed field type of public_use from list_string to list_integer (#560).  Fixes tests to not crash.

### CODE REVIEW

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_core/cgov_core.module [hunk 1]
```diff
@@ -45,6 +45,45 @@ function cgov_core_entity_translation_create(EntityInterface $translation) {
...
+  $query = \Drupal::entityQuery('block_content');
+  $query->condition('info', 'Public Use')
+    ->condition('langcode', 'en');
+  $result = $query->execute();
+  if (count($result)) {
+    $eid = array_shift($result);
+    $block = entity_load('block_content', $eid);
+    $title = $block->info->value . " - " . $langcode;
+    if (isset($public_use[$langcode]) && !$block->hasTranslation($langcode)) {
+      $block_translated = $block->addTranslation('es', [
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/modules/custom/cgov_core/cgov_core.module]: This should not be hard coded to spanish here.

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/templates/content/article/node--cgov-article--full.html.twig [hunk 2]
```diff
@@ -17,5 +17,9 @@
     <footer class="article-footer">
       {{ content.field_related_resources }}
       {{ contentPageMacros.displayDates(node, content) }}
+      {% set catch_cache = content.field_public_use|render %}
+      {% if content.field_public_use|field_raw('value') == '1' %}
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/templates/content/article/node--cgov-article--full.html.twig]: Set the FieldFormatter to be key for field_public_use and remove the catch_cache and set the check to just use field_value.

**Comment by bgnih** [comment]
Per our discussion used `node.field_public_use` instead of `content.field_public_use` and removed public use from entity view modes.

File: docroot/profiles/custom/cgov_site/modules/custom/pdq_cancer_information_summary/config/install/core.entity_view_display.node.pdq_cancer_information_summary.default.yml [hunk 3]
```diff
@@ -120,4 +110,5 @@ content:
     third_party_settings: {  }
     region: content
 hidden:
+  field_public_use: true;
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/modules/custom/pdq_cancer_information_summary/config/install/core.entity_view_display.node.pdq_cancer_information_summary.default.yml]: This broke the tests. removing the semi-colon will fix.

**Comment by bryanpizzillo** [comment]
ODE: http://ncigovcdode43.prod.acquia-sites.com/

**Comment by VictoriaSunNIH** [comment]
@bgnih - Public Use Text is not displaying on Press Releases and Blog Posts.

The block is also missing some content in ENglish (my bad).  Please update the text so that it says:

"If you would like to reproduce some or all of this content, see Reuse of NCI Information for guidance about copyright and permissions. In the case of permitted digital reproduction, please credit the National Cancer Institute as the source and link to the original NCI product using the original product's title; e.g., “<entity/node page title> was originally published by the National Cancer Institute”

**Comment by bgnih** [comment]
@VictoriaSunNIH updates made, should be pushed to the ODE by now.

**Comment by VictoriaSunNIH** [comment]
@lburack - this looks good to me.  Please test on articles, press releases and blog posts.  Thanks!

**Comment by bryanpizzillo** [comment]
It looks like the block is getting cached site wide, so the title only matches the first view. I was on ["Feelings and Cancer"](http://ncigovcdode43.prod.acquia-sites.com/about-cancer/coping/feelings) and saw
screen shot 2019-03-07 at 5 01 24 pm

**Comment by bryanpizzillo** [comment]
It looks like this is a known/abandonded/who knows what issue with token:filter.
https://www.drupal.org/project/token_filter/issues/2743509

**Comment by bgnih** [comment]
Added 'url' cache context to Public Use blocks, fixed form field order.

**Comment by bgnih** [comment]
@VictoriaSunNIH @lburack Better now?

**Comment by lburack** [comment]
I created a press release and an article and their respective titles were correctly displaying in the public use text. But then I changed the title of my published article and pushed it back through the workflow to published. The public use text did not update to reflect the new page title. This does not pass IA review.

**Comment by bgnih** [comment]
Updated caching to clear on node updates, and prevents caching on Preview/Latest draft pages.

**Comment by VictoriaSunNIH** [comment]
This is looking good to me.  Marking this as passed review.

**Comment by bgnih** [comment]
@VictoriaSunNIH Pushing up again with new Develop branch.  See if it looks correct when it deploys.  Here's what I see on Local
image

**Comment by VictoriaSunNIH** [comment]
@bgnih , @lburack - its all looking good now! Marking this as passed.

**Comment by bryanpizzillo** [comment]
I just want to make sure this is working right given the fact that it had to have Adrian's changes merged back in.

**Comment by VictoriaSunNIH** [comment]
I kicked the tires a bit and this is looking good.  Marking as passed review

**Pull request merged**